cupyx/scipy/linalg/special_matrices.py

```diff
@@ -0,0 +1,551 @@
+import math
+import cupy
+from cupy.lib.stride_tricks import as_strided
+
+__all__ = ['tri', 'tril', 'triu', 'toeplitz', 'circulant', 'hankel',
+           'hadamard', 'leslie', 'kron', 'block_diag', 'companion',
+           'helmert', 'hilbert', 'dft',
+           'fiedler', 'fiedler_companion', 'convolution_matrix']
+
+
+def tri(N, M=None, k=0, dtype=None):
+    """ Construct (N, M) matrix filled with ones at and below the kth diagonal.
+    The matrix has A[i,j] == 1 for i <= j + k
+
+    Args:
+        N (int): The size of the first dimension of the matrix.
+        M (int, optional): The size of the second dimension of the matrix. If
+            `M` is None, `M = N` is assumed.
```

Comment:
Please use double-backquotes for the variables. Could you check locally the documentation builds correctly?
```suggestion
            ``M`` is None, ``M = N`` is assumed.
```

Reply:
One of the CI (`docs/readthedocs.org:cupy`) builds the doc for every PR. You can also browse the rendered outcome there.